test/psych/test_array.rb

```diff
@@ -57,6 +57,19 @@ def test_self_referential
       assert_cycle(@list)
     end
 
+    def test_recursive_array_uses_alias
+      @list << @list
+
+      expected = <<~eoyaml
+        --- &1
+        - :a: b
+        - foo
+        - *1
```

Comment:
I don't think we should hard-code the expected YAML.  We may want to change the emitted aliases in the future, and now this test couples the implementation with the exact alias name.  IOW I want to test that aliases _were used_, but I don't care what the aliases were.

Not to mention, we may change indentation in the future etc, so I don't want the test to be coupled to the formatting either.

Reply:
Done, what do you think? [`26af85e` (#568)](https://github.com/ruby/psych/pull/568/commits/26af85e25343c7d1e35d10ba80ceab6efb1cf85c)